src/Selenium2Driver.php

```diff
@@ -492,12 +532,29 @@ public function getScreenshot()
 
     public function getWindowNames()
     {
-        return $this->getWebDriverSession()->window_handles();
+        $origWindow = $this->getWebDriverSession()->window_handle();
+
+        try {
+            $result = array();
+            foreach ($this->getWebDriverSession()->window_handles() as $tempWindow) {
+                $this->getWebDriverSession()->focusWindow($tempWindow);
```

Comment:
isn't there really a better solution - this can be quote slow with a lot of windows

Reply:
Agreed.

But Selenium doesn't have a way of executing JavaScript (for getting a window name) in a window before I switch to it. See https://stackoverflow.com/questions/25760357/get-title-of-window-without-switching-in-selenium-webdriver .

Any ideas?

Reply:
Then any solution switching based on window name seems suboptimal not only because of this, but also because the window names can be non-unique.

User should open a window, store the handle, and then use the handle to switch the window.

Reply:
Do you know any website, that has lots (number of windows, that can cause any slow down of test suite) of opened windows, that is in needed of Selenium-based testing?

Reply:
@mvorisek

1. removing the hack will break backward compatibility
2. it will be (theoretically) slower if the desired window is always at the end
3. a micro-optimization would be to reverse the window handles array, if we know that typically the window we're interested is at the end
4. end-users can make use of `getWindowName()` to be safe (and fast) - as you said in the end

@aik099 we could, however, decide to deprecate usages `windowName` concept and replace it `windowHandle` concept, just to make this clear.

Reply:
I've created https://github.com/minkphp/MinkSelenium2Driver/issues/387 to preserve this discussion upon PR close/merge.

Reply:
Resolving.

All further discussion about using window handles instead of names should happen in the https://github.com/minkphp/Mink/issues/868 .